opencart 1.5.6.4/upload/catalog/controller/smailyforopencart/rss.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class ControllerSmailyForOpencartRss extends Controller {
+
+    public function index() {
+        // Load models.
+        $this->load->model('catalog/product');
+        $this->load->model('tool/image');
+        $this->load->model('localisation/currency');
+
+        // Get params from url for filtering products.
+        if (isset($this->request->get['category'])) {
+            $category = $this->request->get['category'];
+        } else {
+            $category = null;
+        }
+        if (isset($this->request->get['limit'])) {
+            $limit = (int) $this->request->get['limit'];
+        } else {
+            $limit = 50;
+        }
+        // Xml header.
+        $this->response->addHeader('Content-Type: text/xml; charset=utf-8');
+        // Store url.
+        if (isset($this->request->server['HTTPS']) && (($this->request->server['HTTPS'] == 'on') || ($this->request->server['HTTPS'] == '1'))) {
+            $this->data['store_url'] = $this->config->get('config_ssl');
+        } else {
+            $this->data['store_url'] = $this->config->get('config_url');
+        }
+        // Build date.
+        $this->data['last_build_date'] = date('D, d M Y H:i:s');
+        // Currency symbol.
+        $this->data['currency'] = $this->session->data['currency'];
+        // Filter for query.
+        $filter = array(
+            'sort' => 'p.date_added',
+            'order' => 'DESC',
+            'start' => 0,
+            'limit' => $limit
+        );
+        // Add category if set.
+        if ($category) {
+            $filter['filter_category_id'] = $this->getCategoryIdByName($category);
+            $filter['filter_sub_caegory'] = true;
+        }
+        // Get items based on filer.
+        $products = $this->model_catalog_product->getProducts($filter);
+        // Items array for template.
+        $items = [];
+        foreach ($products as $product) {
+            $item = [];
+            // Title.
+            $item['title'] = $product['name'];
+            // Link, guid
+            $item['link'] = $this->url->link('product/product', 'product_id=' . $product['product_id']);
+            // Created date.
+            $item['pubDate'] = $product['date_available'];
```

Comment:
Is `date_available` formatted as RSS requires for `pubDate`?

Reply:
I did `$item['pubDate'] = date('r', strtotime($product['date_available']));`
As I understand RSS requires RFC822, so this converts it to the standard.